### PR TITLE
WIP: More exercise checks

### LINKED ISSUE #65: raindrops: more checks
* Suggest comparing string with `""` rather than checking `len(s)` (makes code clearer)

* Suggest `var s string` rather than `s := ""`

### PR DESCRIPTION
**Pull request by bitfield**
Fixes #65, #66, #68.

### CODE REVIEW

File: track/raindrops/raindrops.go [hunk 1]
```diff
@@ -238,3 +239,22 @@ func examBytesBuffer(pkg *astrav.Package, r *extypes.Response) {
 		r.AppendImprovement(tpl.StringsBuilder.Format("bytes.Buffer"))
 	}
 }
+
+func examCompareLen(pkg *astrav.Package, r *extypes.Response) {
```
Comment (bitfield) [inline track/raindrops/raindrops.go]: I'm flailing around a bit with `astrav` at the moment. Is there a less laborious way of finding whether `len()` is called on a string?

Comment (bitfield) [reply track/raindrops/raindrops.go]: Oh, that's excellent, thanks. I'm slowly getting the hang of things, mostly by copying your existing code. I wonder if it wouldn't be a good idea to have a little tutorial for Exalysis contributors, showing how to use `astrav` to find a few different types of code issues?

Comment (tehsphinx) [reply track/raindrops/raindrops.go]: astrav is still very much WIP. All it does is putting a simpler layer on top of the ast packages of Go.
The key functionality it provides is moving around in a parent/child manner and finding things in a parent. Most of the functionality can be seen the [Node](https://github.com/tehsphinx/astrav/blob/master/node.go#L41) interface. Access to special node type functionality and the underlying ast nodes methods and values can be gained by casting the node to it's actual astrav type.

That's it basically.

Comment (tehsphinx) [reply track/raindrops/raindrops.go]: Note: my mac broke... so currently I'm kind of cut off. 😢

**Comment by tehsphinx** [comment]
Is this ready to be merged?

File: track/raindrops/raindrops.go (hunk 1 above)
Comment (bitfield) [reply track/raindrops/raindrops.go]: Oh dear! That's bad!

**Comment by bitfield** [comment]
Still WIP, but feel free to add to this if you want (as and when your Mac is fixed).

**Comment by tehsphinx** [comment]
@bitfield: any news on this?

**Pull request closed without merging**